compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -509,6 +517,64 @@ encodeModuleRule =
         let (hash, bs) = SS.encodeModule lfVersion m
         return ([], Just (mconcat $ hash : map fst encodedDeps, bs))
 
+-- hlint
+
+-- We need to provide hlint the path to the directory of a file
+-- containing a base configuration file called 'hlint.yaml'.
+-- We test with one version of 'hlint.yaml' and deploy with another
+-- (at the current time, there are good reasons for this). The code
+-- below (which relies on the behavior of `package_app` deploying to
+-- '~/.daml/sdk/x.x.x/damlc/resources/daml-ide-core/data') selects one
+-- or the other depending on context.
+getHlintDataDir :: IO FilePath
+getHlintDataDir = do
+  root <- locateRunfiles $
+    mainWorkspace </> "compiler/damlc/daml-ide-core"
+  let test = root </> "data/test"
+      prod = root </> "daml-ide-core/data/prod"
```

Comment:
That looks wrong. Those files are located next to each other. Why do you try to find `prod` in `compiler/damlc/daml-ide-core/daml-ide-core`.

Reply:
Trust me, it took me and Andreas more than an hour or two to work it out. This is what `package_app` does : it puts the prod version in `daml-ide-core/resources/data/daml-ide-core/data/prod` whereas when going for the test version in the unit tests, it's found in just `data/test`. Go figure.

Reply:
If it does that, then we should fix `package_app` not hack around here. This is almost certainly not going to work when you get it via Bazel runfiles.

Reply:
Well, suppose we fix `package_app` so that its then found via `data/prod`. If we keep this mechanism (and it looks like not one way or the other), then we'd need a different way of sniffing out if we are running in test or prod mode.

Reply:
> not going to work when you get it via Bazel runfiles.
Not sure what you mean here. We are using runfiles aren't we? That is, this is working (or at least some of the time) isn't it?